### PR TITLE
Heater needs Air 

### PR DESCRIPTION
**Pull request by Dormanil**
I rebased @abackwood's changes back onto the most recent master and reactivated the heater's power usage.

### CODE REVIEW

**Comment by quill18** [comment]
No obvious errors noted here.

**Comment by Dormanil** [comment]
Rebased it yet again, now we know for sure that it passes the CI test.

File: Assets/StreamingAssets/LUA/Furniture.lua [hunk 1]
```diff
@@ -562,15 +562,24 @@ end
...
+    end
+    
+    tile = furniture.tile
+    pressure = tile.Room.GetGasPressure() / tile.Room.GetSize()
+    efficiency = ModUtils.Clamp01(pressure / furniture.Parameters["pressure_threshold"].ToFloat())
+    temperatureChangePerSecond = furniture.Parameters["base_heating"].ToFloat() * efficiency
+    temperatureChange = temperatureChangePerSecond * deltaTime
+    
+    World.Current.temperature.ChangeTemperature(tile.X, tile.Y, temperatureChange)
+    ModUtils.ULogChannel("Temperature", "Heat change: " .. temperatureChangePerSecond .. " => " .. World.current.temperature.GetTemperature(tile.X, tile.Y))
```
Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: This log is spamming the console and slowing down the game a lot. Is it still necessary?

**Comment by sboigelot** [comment]
I'm also getting a few warning that seems related:

**Comment by Dormanil** [comment]
Tell me more about the warnings and I might know what to do against it, @sboigelot.

**Comment by sboigelot** [comment]
@Dormanil I made a mistake about the pressure door one, it's on master too sorry.

I will have a look at the temperature warning

**Comment by sboigelot** [comment]
@Dormanil you can reproduce the issue by building a heater outside

Yep, something is wrong with your temperature: NaN.
UnityEngine.Debug:LogWarning(Object)
UberLogger.Logger:ForwardToUnity(Object, LogSeverity, Object, Object[]) (at Assets/UberLogger/UberLogger.cs:505)
UberLogger.Logger:Log(String, Object, LogSeverity, Object, Object[]) (at Assets/UberLogger/UberLogger.cs:452)
Debug:ULogWarningChannel(String, String, Object[]) (at Assets/UberLogger/Debug.cs:152)
Temperature:IsWithinTemperatureBounds(Single) (at Assets/Scripts/Models/Temperature.cs:216)
Temperature:ChangeTemperature(Int32, Int32, Single) (at Assets/Scripts/Models/Temperature.cs:154)
System.Runtime.CompilerServices.ExecutionScope:lambda_method(ExecutionScope, Object, Object[])
MoonSharp.Interpreter.Interop.MethodMemberDescriptor:Execute(Script, Object, ScriptExecutionContext, CallbackArguments)
MoonSharp.Interpreter.Interop.OverloadedMethodMemberDescriptor:PerformOverloadedCall(Script, Object, ScriptExecutionContext, CallbackArguments)
MoonSharp.Interpreter.Interop.<>c__DisplayClass33_0:<GetCallback>b__0(ScriptExecutionContext, CallbackArguments)
MoonSharp.Interpreter.CallbackFunction:Invoke(ScriptExecutionContext, IList`1, Boolean)
MoonSharp.Interpreter.Execution.VM.Processor:Internal_ExecCall(Int32, Int32, CallbackFunction, CallbackFunction, Boolean, String, DynValue)
MoonSharp.Interpreter.Execution.VM.Processor:Processing_Loop(Int32)
MoonSharp.Interpreter.Execution.VM.Processor:Call(DynValue, DynValue[])
MoonSharp.Interpreter.Script:Call(DynValue, DynValue[])
MoonSharp.Interpreter.Script:Call(DynValue, Object[])
MoonSharp.Interpreter.Script:Call(Object, Object[])
LuaUtilities:CallFunction(String, Object[]) (at Assets/Scripts/Utilities/LuaUtilities.cs:40)
FurnitureActions:CallFunctionsWithFurniture(String[], Furniture, Single) (at Assets/Scripts/Models/FurnitureActions.cs:67)
EventAction:Trigger(String, Furniture, Single) (at Assets/Scripts/Models/EventAction.cs:134)
<RegisterSinkOrSource>c__AnonStorey12:<>m__22(Single) (at Assets/Scripts/Models/Temperature.cs:109)
Temperature:ProgressTemperature(Single) (at Assets/Scripts/Models/Temperature.cs:260)
Temperature:Update() (at Assets/Scripts/Models/Temperature.cs:99)
World:Tick(Single) (at Assets/Scripts/Models/World.cs:183)
WorldController:Update() (at Assets/Scripts/Controllers/WorldController.cs:165)

**Comment by sboigelot** [comment]
This is in the IsWithinTemperatureBounds of Temperature.cs
You should probably forbit building the heater outside by adding the tag 
    <TypeTag>OutdoorOnly</TypeTag>
Or better, avoid this issue in from within your lua code.

**Comment by sboigelot** [comment]
testing it again :)

**Comment by sboigelot** [comment]
@Dormanil Nice work. I would extend this feature with a new PR if I were you: You could add an icon to the heater displaying when it doesn't work by lack of pressure. Like the one displayed when it doesn't have power.
I would love to see this, but It's up to you.

**Pull request merged**